src/expr/src/scalar/mod.rs

```diff
@@ -884,6 +884,44 @@ impl MirScalarExpr {
                                     })
                                     .collect(),
                             }
+                        } else if let (
+                            BinaryFunc::Eq,
+                            literal @ MirScalarExpr::Literal(Ok(..), ..),
+                            MirScalarExpr::CallUnary { func, expr },
+                        ) = (&*func, &**expr1, &**expr2)
+                        {
+                            // <lit> = f(<expr>), where f is invertible
+                            //  -->
+                            // <f^-1(lit)> = <expr>
+                            // (if f^-1(lit) doesn't error out)
+                            //
+                            // This enables index usage when an implicit cast was inserted on a
+                            // column reference. E.g., if `a` is smallint, then `a = 5` is
+                            // `smallint_to_integer(a) = 5`, which we now transform to `a = 5`.
+                            if let Some(inverse_func) = func.invert() {
+                                let inved_lit = eval(&MirScalarExpr::CallUnary {
```

Comment:
nit: `inved_lit` ⇒ `inverted_lit` or `lit_inv` is probably more obvoius?

Reply:
changed